org.palladiosimulator.analyzer.slingshot.behavior.spd/src/org/palladiosimulator/analyzer/slingshot/behavior/spd/interpreter/entity/constraint/IntervalConstraintFilter.java

```diff
@@ -0,0 +1,62 @@
+package org.palladiosimulator.analyzer.slingshot.behavior.spd.interpreter.entity.constraint;
+
+import org.palladiosimulator.analyzer.slingshot.behavior.spd.interpreter.entities.FilterObjectWrapper;
+import org.palladiosimulator.analyzer.slingshot.behavior.spd.interpreter.entities.FilterResult;
+import org.palladiosimulator.spd.constraints.policy.IntervalConstraint;
+
+public class IntervalConstraintFilter extends AbstractConstraintFilter<IntervalConstraint> {
+
+	/** The sum of the offset and the interval size. */
+	private final double delta;
+	private final boolean repeat;
+
+	public IntervalConstraintFilter(final IntervalConstraint constraint) {
+		super(constraint);
+		this.delta = constraint.getOffset() + constraint.getIntervalDuration();
+		this.repeat = constraint.isRepeat();
+	}
+
+	@Override
+	public FilterResult doProcess(final FilterObjectWrapper event) {
+		final double currentSimulationTime = event.getEventToFilter().time();
+		if (!this.isWithinInterval(currentSimulationTime)) {
```

Comment:
pretty sure, the boolean expression should not be negated.

Reply:
No, we discussed (and also according to the Docu) the trigger should NOT be active *during the interval*. So if the time is outside the interval, then it's active.

Reply:
oh, i missed that one! sorry!